kube-client/src/client/mod.rs

```diff
@@ -126,7 +126,12 @@ impl Client {
         Self::try_from(Config::infer().await.map_err(Error::InferConfig)?)
     }
 
-    pub(crate) fn default_ns(&self) -> &str {
+    /// Get the default namespace for the client
+    ///
+    /// The namespace is either configured on `context` in the kubeconfig,
+    /// falls back to `default` when running locally,
+    /// or uses the service account's namespace when deployed in-cluster.
+    pub fn default_ns(&self) -> &str {
```

Comment:
```suggestion
    pub fn default_namespace(&self) -> &str {
```

Would want a more readable name if this is to be exposed.
This will require some local compile fixes though. It's used within the client.